### ISSUE TITLE
Kind conformance passes in trait, fails in subclass with same type argument

### ISSUE THREAD

**Issue by scabug** [scala/bug#9337]
```scala
import scala.language.higherKinds

trait TypePreservingFn[T[X <: T[X]]]

trait Validator[T, This <: Validator[T,This]]

trait Foo[T] {

  type `Validator[T]`[This <: Validator[T, This]] = Validator[T, This]

  val f: TypePreservingFn[`Validator[T]`] = ???  // compiles

}

class Bar[T] extends Foo[T] {

  val g: TypePreservingFn[`Validator[T]`] = ???  // does not compile

}
```

**Comment by joroKr21** [comment]
> But I fear my head might explode trying to write the commit comment. Perhaps you have a better chance at that :)

This is what I came up with: scala/scala@1f4611df501c7cd636dcdaecaf61dcf8497dd911
Higher-kinded bounds have found their way again.
Unfortunately it doesn't fix this bug but it does fix #10186

**Comment by joroKr21** [comment]
Here is the issue:
```scala
  trait AliasTypeRef extends NonClassTypeRef {
    require(sym.isAliasType, sym)

    override def dealias    = if (typeParamsMatchArgs) betaReduce.dealias else super.dealias
    override def narrow     = normalize.narrow
    override def prefix     = if (this ne normalize) normalize.prefix else pre
    override def termSymbol = if (this ne normalize) normalize.termSymbol else super.termSymbol
    override def typeSymbol = if (this ne normalize) normalize.typeSymbol else sym
```

the prefix of `Validator[T]` is the prefix of `Validator` instead of `Bar`
